homeassistant/components/matter/climate.py

```diff
@@ -261,6 +262,18 @@ def _update_from_device(self) -> None:
         self._attr_current_temperature = self._get_temperature_in_degrees(
             clusters.Thermostat.Attributes.LocalTemperature
         )
+
+        self._attr_current_humidity = (
+            int(raw_measured_humidity) / HUMIDITY_SCALING_FACTOR
+            if (
+                raw_measured_humidity := self.get_matter_attribute_value(
+                    clusters.RelativeHumidityMeasurement.Attributes.MeasuredValue
+                )
+            )
+            is not None
+            else None
+        )
```

Comment:
I'm not a huge fan of this block. If this pattern is used elsewhere in Matter, it's probably fine, but since `self._attr_current_humidity` should already be initialized with `None` in the base climate component, we don't need to set it to `None` again if we don't get a Matter value.

What do you think about something like this instead?

```python
if (
    humidity_value := self.get_matter_attribute_value(
        clusters.RelativeHumidityMeasurement.Attributes.MeasuredValue
    )
) is not None:
    self._attr_current_humidity = int(humidity_value) / HUMIDITY_SCALING_FACTOR
```

The line wrapping also doesn't look that nice, but we can't really avoid that here.

Reply:
Ah, actually. If the device later reports a `None` value, that would be an issue. Hmm. So, don't use the above suggestion for now.

Reply:
Yeah, let's leave the current variant for now. Sorry for the noise 😄 

PR looks good to me as well